src/webui/abstractwebapplication.cpp

```diff
@@ -433,23 +433,25 @@ bool AbstractWebApplication::isCrossSiteRequest(const Http::Request &request) co
     return true;
 }
 
-bool AbstractWebApplication::validateHostHeader(const Http::Request &request, const Http::Environment &env, const QStringList &domains) const
```

Comment:
The server becomes more and more single-thread design centered.

Reply:
sounds like a bad thing?
Anyway I think the real problem is web server sharing the same event loop with GUI events.

Reply:
What's the problem?
Currently qBittorrent WebUI is just alternative single-user UI. Even if application is built with both GUI and WebUI, I can't imagine they are used simultaneously.